compiler/src/dmd/func.d

```diff
@@ -2199,6 +2199,17 @@ extern (C++) class FuncDeclaration : Declaration
         }
         else
         {
+            foreach (v; closureVars)
+            {
+                if (v.isRef() &&
+                    (global.params.useDIP1000 == FeatureState.enabled && isSafe()))
```

Comment:
Same as https://github.com/dlang/dmd/pull/14363#discussion_r944221525: Don't introduce branches with `global.params.useDIP1000 == FeatureState.enabled`, `-preview=dip1000` is now enabled by default in deprecation mode so use `setUnsafeDIP1000` and/or `previewErrorFunc`.

Reply:
Same as the other one - setUnsafeDIP1000 requires sc, and this function doesn't have sc.

Reply:
I'd save a refectoring of setUnsafeDIP1000 for a future PR.